### ISSUE TITLE
No error for undeclared #private property in `.js` files

### ISSUE THREAD

**Issue by DanielRosenwasser** [microsoft/TypeScript#35962]
```js
class Thing {
    /**
     * @param prop {string}
     */
    constructor(prop) {
        this.#prop = prop;
    }
}
```

This class assigns to  `#prop`, but it's an error to use `#prop` without declaring it.

This is reported as a `SyntaxError`, so this should report an error even outside of `checkJs`.

**Comment by DanielRosenwasser** [comment]
(CC @rjpalmer in case anyone on the team is interested)

**Comment by joeywatts** [comment]
It seems this is an issue for the following cases as well:
- Using a private identifier outside of a class body.
- Using a private identifier in an optional chain.
- Using a private identifier that is not within the lexical scope.

These errors are emitted in the checker, but they need to be emitted at the parse phase (as I understand) to have these errors for JS files which are not being checked.

**Comment by ajafff** [comment]
This is basically a duplicate of #29213. Many errors that should be syntax errors are currently treated as grammar or semantic errors.

**Comment by fatcerberus** [comment]
> it's an error to not use #prop without declaring it.

Do you mean that it’s an error to *use it* without declaring it?  This statement confuses me.

**Comment by sandersn** [comment]
Fixed by #35987

**Comment by DanielRosenwasser** [comment]
No, that's a different issue.

**Comment by sandersn** [comment]
I don't think this should be an error in JS without checkJs on. We don't report errors for assignment to const either, which is another error that can be lexically determined as well.

**Comment by RyanCavanaugh** [comment]
This is fixed. With `--allowJs` and no `checkJs`, TypeScript 3.8.0-dev.20200108 accepts the original `test.js` without diagnostics. The current native compiler and `typescript@next` (7.1.0-dev.20260826.1) report:

> `test.js(6,14): error TS1111: Private field '#prop' must be declared in an enclosing class.`
